Makefile

```diff
@@ -6,9 +6,14 @@ coverage:  ## Run tests with coverage
 	coverage run -m pytest -ra
 	coverage report -m
 
-deps:  ## Install dependencies
-	pip install --upgrade pip
-	pip install -r requirements.txt
+back:  ## Install backend dependencies in a virtualenv - start server
+	(cd backend; python3 -m venv venv; . venv/bin/activate; \
+		pip install -r requirements.txt; python3 src/app.py &)
+
+front: ## Install frontend dependencies in a virtualenv - start server
+	(cd frontend; python3 -m venv venv; . venv/bin/activate; \
+		pip install -r requirements.txt; pip install -r requirements-pre.txt; \
+		streamlit run app.py &)
```

Comment:
These are the same commands that we run in the `Dockerfile`. It could be useful to have a `setup.py` script (one for backend, one for frontend) that we run from both the `Dockerfile` and the `Makefile`. The difference is that for now we don't run the Docker app in a virtual environment, but maybe it's not a problem to do it. What do you think @Rubix982 ?

Reply:
This could be considered.

Reply:
The `Dockerfile` seems almost the same for me except these lines,

```Dockerfile
# pip does not like being run as root, so create a user
RUN useradd --create-home jina

# Add the data folder locally to the container
COPY . .

# Switch to user
USER jina

# Path change needed for huggingface-cli and jina
ENV PYTHONPATH "${PYTHONPATH}:/home/jina/.local/bin"
ENV PATH "${PATH}:/home/jina/.local/bin"
```

We would be creating a `jina` on the host machine, which would require sudo permissions on the user's machine. Feels not right to ask for `sudo` permissions to build a container, and I'm not sure how I convert `COPY. .` to as bash commands. The containers run only two commands,

```Dockerfile
# Install dependencies
RUN pip3 install -r requirements.txt
RUN pip3 install -r requirements-pre.txt
```

And does not really deal with a virtual env. Can't think of an elegant solution for this. :S

Reply:
Well, maybe there's no need to complicate things right now. Let's keep it like this and move on.